NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -121,64 +135,70 @@ public void run() {
 		}
 	}
 	
+	public static List<Group> getSubGroups(String name) {
+		List<Group> groups = groupManagerDao.getSubGroups(name);
+		for (Group group : groups) {
+			groupsByName.put(group.getName().toLowerCase(), group);
+			groupsById.put(group.getGroupId(), group);
+		}
+		return groups;
+	}
+	
 	/*
 	 * Making this static so I can use it in other places without needing the GroupManager Object.
 	 * Saves me code so I can always grab a group if it is already loaded while not needing to check db.
 	 */
-	public static Group getGroup(String groupName){
-		groupName = groupName.toLowerCase();
-		if (groupsByName.containsKey(groupName))
-			return groupsByName.get(groupName);
-		else{ 
-			Group group;
-			group = groupManagerDao.getGroup(groupName);
+	public static Group getGroup(String name){
+		String lower = name.toLowerCase();
+		if (groupsByName.containsKey(lower)) {
+			NameLayerPlugin.log(Level.INFO, "Group " + name + " found");
+			return groupsByName.get(lower);
+		} else { 
+			NameLayerPlugin.log(Level.INFO, "Group " + name + " not found, looking in DB");
+			Group group = groupManagerDao.getGroup(name);
 			if (group != null) {
-				groupsByName.put(groupName, group);
+				groupsByName.put(lower, group);
 				groupsById.put(group.getGroupId(), group);
 			}
 			return group;
 		}
 	}
-	
+		
 	public static Group getGroup(int groupId){
-		if (groupsById.containsKey(groupId))
+		if (groupsById.containsKey(groupId)) {
 			return groupsById.get(groupId);
-		else{ 
-			Group group;
-			group = groupManagerDao.getGroup(groupId);
+		} else { 
+			Group group = groupManagerDao.getGroup(groupId);
 			if (group != null) {
-				int id = group.getGroupId();
-				if (groupsByName.containsKey(group.getName())) {
```

Comment:
That might have been what was causing another caching issue we were dealing with, good call to change it.